src/main/webapp/util/Validator.js

```diff
@@ -135,6 +136,25 @@ class Validator {
         return false;
     }
 
+    FileValidator(field, validator, data) {
+        if(data) {
+            const { isValidExtention, fileSize } = parseFileValidator(data, validator.supportedFileTypes);
+            if(!isValidExtention) {
+                return {
+                    errorField: field,
+                    errorMsg: validator.errorMsg || getFormattedMessage(24, [validator.supportedFileTypes])
+                };
+            }
+            if(fileSize > 500000) {
```

Comment:
Let's have it as a constant somewhere so we don't have magic numbers in the code.

And we can reuse that in the messageDict.ha file as well, instead of hardcoding it there.